File: Arrays_n_Strings/min_start_value.js
function minStartValue(nums) {
    let startValue = 1;  // Initial startValue
    let runningSum = startValue;

    for (let num of nums) {
        runningSum += num;
        // If runningSum is less than 1, adjust startValue
        if (runningSum < 1) {
            startValue += 1 - runningSum;
            runningSum = 1;
        }
    }

    return startValue;
}

// Test the function with provided examples
console.log(minStartValue([-3, 2, -3, 4, 2]));  // Output: 5
console.log(minStartValue([1, 2]));             // Output: 1
console.log(minStartValue([1, -2, -3]));        // Output: 5

File: Arrays_n_Strings/min_start_value.py
def minStartValue(nums):
    startValue = 1  # Initial startValue
    running_sum = startValue

    for num in nums:
        running_sum += num
        # If running_sum is less than 1, adjust startValue
        if running_sum < 1:
            startValue += 1 - running_sum
            running_sum = 1

    return startValue

# Test the function with provided examples
print(minStartValue([-3, 2, -3, 4, 2]))  # Output: 5
print(minStartValue([1, 2]))             # Output: 1
print(minStartValue([1, -2, -3]))        # Output: 5
